lib/upgrade/etcd/conversionv1v3/rule.go

```diff
@@ -166,15 +166,16 @@ func rulebackendToAPIv3(br model.Rule) apiv3.Rule {
 	notSrcSelector := mergeTagsAndSelectors(br.NotSrcSelector, br.NotSrcTag)
 	notDstSelector := mergeTagsAndSelectors(br.NotDstSelector, br.NotDstTag)
 
-	var v3Protocol numorstring.Protocol
+	var v3Protocol *numorstring.Protocol
 	if br.Protocol != nil {
-		v3Protocol = numorstring.ProtocolV3FromProtocolV1(*br.Protocol)
+		protocol := numorstring.ProtocolV3FromProtocolV1(*br.Protocol)
+		v3Protocol = &protocol
 	}
 
 	return apiv3.Rule{
 		Action:      ruleActionV1ToV3API(br.Action),
 		IPVersion:   br.IPVersion,
-		Protocol:    &v3Protocol,
+		Protocol:    v3Protocol,
```

Comment:
Looks like the NotProtocol field is also missing the protocol conversion logic.

Reply:
As it is written I don't think it has the same problem because the ProtocolV3FromProtocolV1 is done on that field, should that be done there too?

Reply:
Looks like it should have the conversion, I'll add that in.

Reply:
Ok I've added it to NotProtocol too